src/main/scala/io/iohk/ethereum/jsonrpc/EthProofService.scala

```diff
@@ -0,0 +1,158 @@
+
+  /**
+    * Get account and storage values for account including Merkle Proof.
+    *
+    * @param address address of the account
+    * @param storageKeys storage keys which should be proofed and included
+    * @param blockNumber block number or string "latest", "earliest"
+    * @return
+    */
+  def run(
+      address: Address,
+      storageKeys: Seq[StorageProofKey],
+      blockNumber: BlockParam
+  ): Task[Either[JsonRpcError, ProofAccount]] = Task {
+    for {
+      blockNumber <- resolveBlock(blockNumber).map(_.block.number)
+      account <- Either.fromOption(
+        blockchain.getAccount(address, blockNumber),
+        noAccount(address, blockNumber)
+      )
+      accountProof <- Either.fromOption(
+        blockchain.getAccountProof(address, blockNumber).map(_.map(asRlpSerializedNode)),
+        noAccountProof(address, blockNumber)
+      )
+      storageProof <- getStorageProof(account, storageKeys)
+    } yield mkAccountProof(account, accountProof, storageProof, address)
+  }
+
+  def getStorageProof(
+      account: Account,
+      storageKeys: Seq[StorageProofKey]
+  ): Either[JsonRpcError, Seq[StorageProof]] = {
+    storageKeys.toList
+      .map { storageKey =>
+        blockchain
+          .getStorageProofAt(
+            rootHash = account.storageRoot,
+            position = storageKey.v,
+            ethCompatibleStorage = ethCompatibleStorage
+          )
+          .map { case (value, proof) => StorageProof(storageKey, value, proof.map(asRlpSerializedNode)) }
+          .toRight(noStorageProof(account, storageKey))
+      }
+      .sequence
+      .map(_.toSeq)
+  }
+
+  private def noStorageProof(account: Account, storagekey: StorageProofKey): JsonRpcError =
+    JsonRpcError.LogicError(s"No storage proof for [${account.toString}] storage key [${storagekey.toString}]")
+
+  private def noAccount(address: Address, blockNumber: BigInt): JsonRpcError =
+    JsonRpcError.LogicError(s"No storage proof for Address [${address.toString}] blockNumber [${blockNumber.toString}]")
+
+  private def noAccountProof(address: Address, blockNumber: BigInt): JsonRpcError =
+    JsonRpcError.LogicError(s"No storage proof for Address [${address.toString}] blockNumber [${blockNumber.toString}]")
+
+  private def mkAccountProof(
+      account: Account,
+      accountProof: Seq[ByteString],
+      storageProof: Seq[StorageProof],
+      address: Address
+  ): ProofAccount =
+    ProofAccount(
+      address = address,
+      accountProof = accountProof,
+      balance = account.balance,
+      codeHash = account.codeHash,
+      nonce = account.nonce,
+      storageHash = account.storageRoot,
+      storageProof = storageProof
+    )
+
+  private def asRlpSerializedNode(node: MptNode): ByteString =
+    ByteString(MptTraversals.encodeNode(node))
+
+  private def resolveBlock(blockParam: BlockParam): Either[JsonRpcError, ResolvedBlock] = {
```

Comment:
Yes. We spend today some time to reuse BlockResolver to replace resolveBlock in EthProofService and EthService.
It turned out it will require lots of changes in EthServiceSpec & will lead to massive conflicts. So with @AnastasiiaL @bsuieric we decided to allow the duplication and get back to this when EthService will be splitted.